pkg/workceptor/kubernetes.go

```diff
@@ -351,94 +356,114 @@ func (kw *KubeUnit) KubeLoggingWithReconnect(streamWait *sync.WaitGroup, stdout
 			// fail to get pod, no need to continue
 			return
 		}
+		prevPodDelay, curPodDelay = 1, 1
 
 		logStream, err := kw.kubeLoggingConnectionHandler(true, sinceTime)
 		if err != nil {
 			// fail to get log stream, no need to continue
 			return
 		}
+		defer logStream.Close()
 
 		// read from logstream
 		streamReader := bufio.NewReader(logStream)
-		for *stdinErr == nil { // check between every line read to see if we need to stop reading
+		// TODO: removed stdinErr, not set in the loop, doesn't change after first check above
+		for { // check between every line read to see if we need to stop reading
 			line, err := streamReader.ReadString('\n')
 			if err != nil {
-				if kw.GetContext().Err() == context.Canceled {
-					kw.GetWorkceptor().nc.GetLogger().Info(
-						"Context was canceled while reading logs for pod %s/%s. Assuming pod has finished",
+				// First check if the error is not EOF, if error is not EOF set error and mark the job as failed.
+				if err != io.EOF {
```

Comment:
I do not think we should immediately fail the job if we get an error (non-EOF) attempting to read a line from the log stream.

We now know that we'll EOFs for jobs that are still running, but I think it's a mistake to interpret any non-EOF error as a failed job. I think we were retrying those before, right?